src/rust/engine/process_execution/src/cache.rs

```diff
@@ -83,31 +109,44 @@ impl CommandRunner {
     fingerprint: Fingerprint,
     context: Context,
   ) -> impl Future<Item = Option<FallibleExecuteProcessResultWithPlatform>, Error = String> {
+    use bazel_protos::remote_execution::ExecuteResponse;
     let file_store = self.file_store.clone();
+
+    let platform_map = self.platform_map.clone();
+
     self
       .process_execution_store
-      .load_bytes_with(fingerprint, |bytes| {
-        let mut execute_response = bazel_protos::remote_execution::ExecuteResponse::new();
+      .load_bytes_with(fingerprint.clone(), move |bytes| {
+        let mut execute_response = ExecuteResponse::new();
         execute_response
           .merge_from_bytes(&bytes)
           .map_err(|e| format!("Invalid ExecuteResponse: {:?}", e))?;
-        Ok(execute_response)
-      })
-      .and_then(move |maybe_execute_response| {
-        if let Some(execute_response) = maybe_execute_response {
-          crate::remote::populate_fallible_execution_result(
-            file_store,
-            execute_response,
-            vec![],
-            context.workunit_store,
-            Platform::current().unwrap(), //TODO this is incorrect and will be fixed in a follow-up PR
-          )
-          .map(Some)
-          .to_boxed()
-        } else {
-          future::ok(None).to_boxed()
-        }
+
+        let platform_lookup_result = platform_map
+          .lock()
+          .get(&fingerprint)
+          .cloned()
+          .ok_or_else(|| "Execution platform not found in store".to_string())?;
```

Comment:
As I read this, this means that any time you're not on the same pantsd instance as where the process actually ran, you'll get this instead of the platform, right?

That seems bad... If there's a reason to need this information, we probably need to _always_ provide it.

Some ways I could see for us to do this, with varying trade-offs, in my rough preferred order:
* Propose a new field to the `ExecutedActionMetadata` [field](https://github.com/bazelbuild/remote-apis/blob/1b16ed76965afa8bb229ba22b90745c546416443/build/bazel/remote/execution/v2/remote_execution.proto#L792) which optionally notes the execution platform of the worker. That would also mean our code could stop hard-coding that that platform is Linux.
* Define a wrapper type, either a protobuf type or a Rust type which we teach to serialize/deserialize with serde, which has a platform and an `ExecuteResponse`. Store that instead of just the `ExecuteResponse` in our lmdb.
* Use an extension field on `ExecuteResponse` to set the platform. Reset the field before we do anything else with the deserialized value (to avoid accidentally missing caches). That would be kind of hacky, because the standard proto may start using that field number at some point.